src/knex.ts

```diff
@@ -0,0 +1,21 @@
+import { knex as _knex, Knex } from 'knex'
+import env from './env'
+
+export default class SingletonKnex {
+	private static instance: Knex
+	private constructor() {
+		//
+	}
+	public static getInstance(): Knex {
+		if (!env.DATABASE_URI) {
+			throw new Error('DATABASE_URI variable not found')
+		}
+		if (!SingletonKnex.instance) {
+			SingletonKnex.instance = _knex({
+				client: 'pg',
+				connection: env.DATABASE_URI
+			})
+		}
+		return SingletonKnex.instance
+	}
+}
```

Comment:
```suggestion
let instance: Knex | undefined

export default function knex() {
	if (!env.DATABASE_URI) {
		hrow new Error('DATABASE_URI variable not found')
	}
	if (instance == undefined) {
		instance =  _knex({
			client: 'pg',
			connection: env.DATABASE_URI
		})
	}
	return instance
}
```

Reply:
Yes, this can be done both with class and function. whichever you prefer.